music21/romanText/tsvConverter.py

```diff
@@ -850,6 +928,66 @@ def getForm(rn: roman.RomanNumeral) -> str:
     return ''
 
 
+def handleAddedTones(dcml_chord: str) -> str:
+    '''
+    Converts DCML added-tone syntax to music21.
+
+    >>> romanText.tsvConverter.handleAddedTones('V(64)')
+    'Cad64'
+
+    >>> romanText.tsvConverter.handleAddedTones('i(4+2)')
+    'i[no3][add4][add2]'
+
+    >>> romanText.tsvConverter.handleAddedTones('Viio7(b4)/V')
+    'Viio7[no3][addb4]/V'
+
+    When in root position, 7 does not replace 8:
+    >>> romanText.tsvConverter.handleAddedTones('vi(#74)')
+    'vi[no3][add#7][add4]'
+
+    When not in root position, 7 does replace 8:
+    >>> romanText.tsvConverter.handleAddedTones('ii6(11#7b6)')
+    'ii6[no8][no5][add11][add#7][addb6]'
+
+
+    '''
+    m = re.match(
+        r'(?P<primary>.*?(?P<figure>\d*(?:/\d+)*))\((?P<added_tones>.*)\)(?P<secondary>/.*)?',
+        dcml_chord
+    )
+    if not m:
+        return dcml_chord
+    primary = m.group('primary')
+    added_tones = m.group('added_tones')
+    secondary = m.group('secondary') if m.group('secondary') is not None else ''
+    figure = m.group('figure')
+    if primary == 'V' and added_tones == '64':
+        return 'Cad64' + secondary
+    added_tone_tuples: t.List[t.Tuple[str, str, str, str, str]] = list(
+        # after https://github.com/johentsch/ms3/blob/main/src/ms3/utils.py
+        re.findall(r"((\+|-)?(\^|v)?(#+|b+)?(1\d|\d))", added_tones)
```

Comment:
single quotes.  

What is the license on the ms3/utils.py file?  If not BSD, need permission to reuse -- it's probably within the realm of fair use, but I don't want any claims from the GPL community that would make it a problem.

Reply:
It's GNU v3. The only thing taken from @johentsch's code is the regex, which I then altered and will in fact alter further since we are not using the group that captures the entire match. Perhaps simplest solution to your concerns is just to remove the attribution? Or clarify "regex modeled on URL"?

Reply:
With GNU I want to be extra cautious.  How about rewriting it with re.VERBOSE which would be a lot easier anyhow, and then sufficiently derived as not to have license issues:

```
re.findall(
    r'''(
          (\+|-)?   # front alterations, or whatever this is.
          (\^|v)?   # what is this?
          (#+|b+)?    # sharps or flats
          (1\d|\d)   # numbers 0-19, in practice, 1-13
    )''', added_tones, re.VERBOSE)
```

That would be REALLY helpful for reading later.
re.VERBOSE is the best way to make a complex
regular expression understandable and debuggable.

I should use this more.